test/params_api_test.c

```diff
@@ -26,7 +26,15 @@ static void swap_copy(unsigned char *out, const void *in, size_t len)
         out[j] = ((unsigned char *)in)[len - j - 1];
 }
 
-static void copy_to_le(unsigned char *out, const void *in, size_t len)
+/*
+ * A memory copy that converts the native byte ordering either to or from
+ * little endian format.
+ *
+ * On a little endian machine copying either is just a memcpy(3), on a
+ * big endian machine copying from native to or from little endian involves
+ * byte reversal.
+ */
+static void le_copy(unsigned char *out, const void *in, size_t len)
```

Comment:
I think this function needs to know the length of the native type and the length of the little-endian side.
and if the type is sign-extended of zero-extended

Reply:
This is what I've tried to avoid.

(apologies for being a bit abrupt, I had my first physio session today and I'm hurting)